llvm-bazel/llvm-project-overlay/mlir/BUILD

```diff
@@ -3144,18 +3137,15 @@ cc_library(
         ":StandardOps",
         ":Support",
         "//llvm:Support",
-        "//third_party/vulkan_headers",
-        "//third_party/vulkan_loader",
+        "@vulkan_headers",
+        "@vulkan_sdk//:sdk"
```

Comment:
Can you create something that is `select`-able on, and make the entire rule (and the dependencies on the rule below) conditioned on Vulkan SDK being available? Something like: https://stackoverflow.com/questions/63677527/conditionally-create-a-bazel-rule-based-on-config

As it is, if you touch this BUILD file (even to query) without the SDK, it'll fail. =/ This seems especially bad for users of LLVM that aren't *specifically* interested in MLIR but now a pattern that just happens to include the directory will cause build errors. Unless I'm missing something?

Reply:
That's not quite right. To query this rule, you don't even need `@vulkan_sdk` defined. To build (or cquery) this rule, you need it defined, but it doesn't actually have to do anything. To run the things that depend on this rule, you need to actually have the Vulkan SDK or you'll get link errors. A nice feature of that is that you can build things even without having the Vulkan SDK. Does that seem acceptable? Or do you want this rule to not even be defined if the vulkan_sdk isn't available. Or we could have it select to only add the vulkan_sdk if it's available, which would have the same feature of being able to build without it. I can look into that as well.

Another option would be to leave this rule (and any that depends on an external repo alias as `manual`). Then they won't get picked up in wildcard builds so no one should care and they don't have to have `@vulkan_sdk` defined at all. The disadvantage is that `bazel build @llvm-project//...` no longer actually builds everything, we'd need to pipe query to build/test, although piping query to test is actually the only way to get bazel to build and test everything in a single command (just `bazel test` won't build binaries unused in tests). In that case we'll need to find another way to tag the rules currently tagged manual to differentiate them as not buildable.

Reply:
Ideally, we'd disable this rule (either by marking it manual or something else) and disable dependencies on this rule when the Vulcan SDK isn't installed. Building or depending on it will only result in a (late) error.

I'd like to be able to build as much as possible without the SDK installed. Not a big deal if the rule is gone or manual though, as long as when the SDK *is* installed, it comes back automatically?

Reply:
If we disable it when you don't have the SDK, then you can't build it even if you want to test it building and don't have the SDK installed. As it stands, you don't need the SDK installed to build it.

Reply:
I'm fine with this for now, but I'd still like a better way for consumers of the Bazel rules to avoid accidentally ending up downloading and building against the headers.

As one user who is especially concerned with things like licenses of dependencies (much like LLVM itself is), I would really like a way to ensure that *no* dependencies other than llvm-bazel and the llvm-project are fetched in my Bazel build. Maybe that's a weird desire, but I'd like to see if there is a reasonably simple way to support that purely for simplicity of humans reasoning about these things. =] Anyways, fine to explore this in follow-ups, especially given the laziness.

Reply:
On a project level, this is super easy. Just don't configure a repo rule for these dependencies. The WORKSPACE file for llvm-bazel isn't used by dependent projects. That seems much more robust than whether the SDK happens to be installed in terms of making sure you don't accidentally mix licenses.